### ISSUE TITLE
您好，请问您这篇文章在哪里可以搜到呢

### ISSUE THREAD

**Issue by zzidlezz** [tswang0116/TA-DCH#2]

**Comment by tswang0116** [comment]
目前还在投稿，录用后将会公开

**Comment by zzidlezz** [comment]
请问您的IAPR-TC12这个数据集怎么打开呢，我尝试了matlab不能打开，python的f5py的包按照您load_data里的 images = Data['IAll'][:]
    labels = Data['LAll'][:]
    tags = Data['TAll'][:]这样也打不开

**Comment by tswang0116** [comment]
Data = h5py.File(’IAPR-TC12数据集的存放路径‘)
    images = Data['IAll'][:]
    labels = Data['LAll'][:]
    tags = Data['TAll'][:]
如果这样打不开，麻烦把您的错误写一下

**Comment by zzidlezz** [comment]
可以读取了，我自己的问题，您的这些图片224x224是已经预处理过了吗

**Comment by tswang0116** [comment]
没有经过其他处理，只是把图像放缩到224*224了

**Comment by zzidlezz** [comment]
您好，请问您提供的例子里那个DGCPN.pth是用DGCPN模型在本文提供的数据集上训练出来的模型，然后加载到这个代码中的吗，其他攻击的模型也是同理吧

**Comment by tswang0116** [comment]
六种被攻击的跨模态哈希检索方法均在我们所提供的数据集上训练得到，然后按照说明放到相应文件夹中即可

**Comment by zzidlezz** [comment]
您好这篇文章最后可视化的结果，就是给定查询的文本返回前几张图片的代码可以参考下吗

**Comment by tswang0116** [comment]
def calc_hamming_dist(B1, B2):
    q = B2.shape[1]
    if len(B1.shape) < 2:
        B1 = B1.unsqueeze(0)
    distH = 0.5 * (q - B1.mm(B2.transpose(0, 1)))
    return distH

def Result(qB, rB, query_label, retrieval_label, k):
    num_query = query_label.shape[0]
    index = torch.zeros(num_query, k)
    for i in range(num_query):
        hamm = calc_hamming_dist(qB[i, :], rB)
        _, ind = torch.sort(hamm)
        ind.squeeze_()
        index[i] = ind[:k]
    return index
